internal/controller/helmrelease_controller_chart.go

```diff
@@ -118,7 +118,12 @@ func (r *HelmReleaseReconciler) loadHelmChart(source *sourcev1b2.HelmChart) (*ch
 	defer f.Close()
 	defer os.Remove(f.Name())
 
-	artifactURL := source.GetArtifact().URL
+	artifact := source.GetArtifact()
+	if artifact == nil {
+		return nil, fmt.Errorf("source '%s/%s' has no artifact", source.GetNamespace(), source.GetName())
+	}
```

Comment:
Would do this before the creation of the temporary file, as we do not need it without the Artifact being present.